tests/lang/s05_pragmas/s02_misc/t02_flag_pragmas.nim

```diff
@@ -0,0 +1,67 @@
+discard """
+description: '''
+Disable or enable safety checks
+'''
+"""
+
+## The listed pragmas here can be used to override the code 
+## generation options for a proc/method/converter.
+
+block enable_bound_checking:
+  {.push boundChecks:on.}
+  proc impl() = 
+    var gotDefect = false
+    try:
+      var a: array[10, int]
+      let idx = 9
+      assert cast[ptr array[9, int]](a.addr)[idx] == 0
+
+    except IndexDefect:
```

Comment:
Wait, you can catch defects?

Reply:
Yes, it is possible to catch defects just "not recommended" because of the `--panics:on` that someone *might* turn on.